### PR TITLE
Add home directory inputs

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Add optional `home_directory_mappings` key to `sftp_users`
* Add optional `s3_bucket_name` key to `sftp_users`
* Tag everything
* Remove `${Transfer:*}` vars

## why
* Allow overriding user assigned home directory mappings and add additional
* Allow overriding user assigned s3 bucket 
* Fix issue where user and s3 bucket are calculated from the ${Transfer:*} vars
    - e.g. `${Transfer:UserName}` would be applied but then the API would result in `bob` instead of this value so terraform would then try to revert the change.

## references
- Closes https://github.com/cloudposse/terraform-aws-transfer-sftp/issues/20
- Closes https://github.com/cloudposse/terraform-aws-transfer-sftp/issues/25

### CODE REVIEW

File: examples/complete/outputs.tf [hunk 1]
```diff
@@ -1,9 +1,9 @@
 output "id" {
-  description = "ID of the created example"
-  value       = module.example.id
+  description = "The null label ID passed to each resource"
```
[the remark this answers is not in the document]
Comment (nitrocode) [reply examples/complete/outputs.tf:2]: How do you distinguish this from the output `transfer_sftp_id` which is different ?

File: examples/complete/main.tf [hunk 2]
```diff
@@ -7,23 +7,35 @@ provider "awsutils" {
...
+module "sftp" {
   source = "../.."
 
   sftp_users = var.sftp_users
 
-  s3_bucket_name = module.s3_bucket.bucket_id
+  s3_bucket_name = module.s3_bucket["home"].bucket_id
+
+  home_directory_mappings = {
+    "extra" = {
```
[the remark this answers is not in the document]
Comment (nitrocode) [reply examples/complete/main.tf:34]: In order to allow multiple s3 buckets

File: examples/complete/main.tf [hunk 3]
```diff
@@ -7,23 +7,35 @@ provider "awsutils" {
...
+      entry  = "/"
+      target = "/${module.s3_bucket["extra"].bucket_id}/$${Transfer:UserName}"
```
[the remark this answers is not in the document]
Comment (nitrocode) [reply examples/complete/main.tf:36]: extra

File: examples/complete/outputs.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (nitrocode) [reply examples/complete/outputs.tf:2]: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/transfer_server#id

[the remark this answers is not in the document]
Comment (nitrocode) [reply examples/complete/outputs.tf]: I reverted this for now so I can do it in a separate PR

**Review by milldr — APPROVED** [review]
LGTM

**Review by Benbentwo — APPROVED** [review]

**Pull request merged**